wordpress-mobile/WordPress-Android
# Deeplink improvements

This PR ties a few loose ends:
- move app link handling for `wordpress://stats` to the `StatsLinkHandler`
- move app link handling for `wordpress://notifications to the `NotificationsLinkHandler`
- move app link handling for `wordpress://read to the `ReaderLinkHandler`
- Move all the deeplink code into a separate package for better readibility

The only small difference in behaviour is that previously for `wordpress://stats` link we were selecting the primary site and showing stats for that site. For consistency I've changed that to show the stats for the currently selected site (just switching to the `Stats` screen). I think this change makes sense.

To test:
- Make sure you're logged in in the app
- Click on `wordpress://stats` link (in case you don't know how to create one, a link like `public-api.wordpress.com/mbar/?redirect_to=wordpress://stats` works as well
- Select the app to handle the link
- Notice the app opens the stats screen

To test:
- Make sure you're logged in in the app
- Click on `wordpress://notifications` link (in case you don't know how to create one, a link like `public-api.wordpress.com/mbar/?redirect_to=wordpress://notifications` works as well
- Select the app to handle the link
- Notice the app opens the notifications screen

To test:
- Make sure you're logged in in the app
- Click on `wordpress://read` link (in case you don't know how to create one, a link like `public-api.wordpress.com/mbar/?redirect_to=wordpress://read` works as well
- Select the app to handle the link
- Notice the app opens the reader screen

## Regression Notes
1. Potential unintended areas of impact
- other deeplinks

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- I've added unit tests and tested the other links 

3. What automated tests I added (or what prevented me from doing so)
- unit tests

PR submission checklist:
